### DISCUSSION TITLE
Volume jumps to 100 and sometimes back

### DISCUSSION THREAD

**Discussion by cYCL157** [rern/rAudio#1584] [Bug Reports]
I grabbed a screen capture from a chrome tab on my windows 11 desktop to show what happens with the volume.  It's not always as you can see in this.  I use the +R icon (long press) to reload the app twice in this and you can see how it sometimes goes to 100 and stays and sometimes between what I had it set and and 100 and then returns.  I have this same thing happen in my iPhone 15 using Safari.

[screen-capture.webm](https://github.com/rern/rAudio/assets/275680/263807b7-234a-402a-8a1a-080e6a8e1a18)

I'd be happy to gather anything else if needed.

Cort

**Comment by markblinkhorn** [comment]
Similar thing just started happening with rAudio 20240210 on RPi 3B+. Whenever a new playlist is loaded the GUI volume jumps to an indicates 100 but playback continues at the previous volume level - however, operating the GUI control to lower the setting causes the volume to increase before it can be brought back to a reasonable listening level.. Recent versions have maintained the display and apparent volume setting between paylist loads.

**Comment by cYCL157** [comment]
This is still a problem with 20240212 (as well as it was with the one the few days before).

**Comment by markblinkhorn** [comment]
@cYCL157 I concur. I have also run pacman -Syu on my systems (64bit R-Pi 3B+, 36bit R-Pi 3B and 32bit R-Pi 2 Legacy) and the issue remains on all 3. However, I do note that if I ssh to the Pi and run alsamixer to view the volume setting (without changing anything) the GUI will revert back to the correct mixer setting. Also, going to Settings-Player-Output and clicking the speaker icon next to Mixer Device causes the GUI to revert to the correct volume setting - I am using Mixer Device for the Volume Control on all of my installations. @rern does this narrow it down any?

[Edit] TBF, just refreshing the page seems to work the trick too - using a PC based brower at least...

Mark

**Reply by cYCL157** [reply]
yes, i mostly just refresh the page on my desktop -- I have more issues with my iPhone than the desktop though and I have the safari tab saved as a link and end up having to force end that.  It really seems that there is something out of sync with the websocket communication and that when refreshing it doesn't seem to refresh everything.  I notice that when my chrome tab on my desktop has lost focus for a long time or my safari tab loses focus for another app.

**Reply by cYCL157** [reply]
Here is another example of where my desktop tab seems to be out of sync with the actual complete play data -- where the position of the track seems right but the length is wrong:

https://i.imgur.com/cmu7Ksf.png

**Comment by rern** [comment]
Please run command: `amixer` and post the output.

**Reply by cYCL157** [reply]
rAudio:~  #  amixer
Simple mixer control 'DSP Program',0
  Capabilities: enum
  Items: 'FIR interpolation with de-emphasis' 'Low latency IIR with de-emphasis' 'High attenuation with de-emphasis' 'Fixed process flow' 'Ringing-less low lat                                              ency FIR'
  Item0: 'FIR interpolation with de-emphasis'
Simple mixer control 'Analogue',0
  Capabilities: pvolume
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 1
  Mono:
  Front Left: Playback 1 [100%] [0.00dB]
  Front Right: Playback 1 [100%] [0.00dB]
Simple mixer control 'Analogue Playback Boost',0
  Capabilities: volume
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: 0 - 1
  Front Left: 0 [0%] [0.00dB]
  Front Right: 0 [0%] [0.00dB]
Simple mixer control 'Auto Mute',0
  Capabilities: pswitch
  Playback channels: Front Left - Front Right
  Mono:
  Front Left: Playback [on]
  Front Right: Playback [on]
Simple mixer control 'Auto Mute Mono',0
  Capabilities: pswitch pswitch-joined
  Playback channels: Mono
  Mono: Playback [on]
Simple mixer control 'Auto Mute Time Left',0
  Capabilities: enum
  Items: '21ms' '106ms' '213ms' '533ms' '1.07s' '2.13s' '5.33s' '10.66s'
  Item0: '21ms'
Simple mixer control 'Auto Mute Time Right',0
  Capabilities: enum
  Items: '21ms' '106ms' '213ms' '533ms' '1.07s' '2.13s' '5.33s' '10.66s'
  Item0: '21ms'
Simple mixer control 'Clock Missing Period',0
  Capabilities: enum
  Items: '1s' '2s' '3s' '4s' '5s' '6s' '7s' '8s'
  Item0: '1s'
Simple mixer control 'Deemphasis',0
  Capabilities: pswitch pswitch-joined
  Playback channels: Mono
  Mono: Playback [on]
Simple mixer control 'Digital',0
  Capabilities: pvolume pswitch
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 207
  Mono:
  Front Left: Playback 115 [56%] [-46.00dB] [on]
  Front Right: Playback 115 [56%] [-46.00dB] [on]
Simple mixer control 'Max Overclock DAC',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 40
  Mono: 0 [0%]
Simple mixer control 'Max Overclock DSP',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 40
  Mono: 0 [0%]
Simple mixer control 'Max Overclock PLL',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 20
  Mono: 0 [0%]
Simple mixer control 'Volume Ramp Down Emergency Rate',0
  Capabilities: enum
  Items: '1 sample/update' '2 samples/update' '4 samples/update' 'Immediate'
  Item0: '1 sample/update'
Simple mixer control 'Volume Ramp Down Emergency Step',0
  Capabilities: enum
  Items: '4dB/step' '2dB/step' '1dB/step' '0.5dB/step'
  Item0: '4dB/step'
Simple mixer control 'Volume Ramp Down Rate',0
  Capabilities: enum
  Items: '1 sample/update' '2 samples/update' '4 samples/update' 'Immediate'
  Item0: '1 sample/update'
Simple mixer control 'Volume Ramp Down Step',0
  Capabilities: enum
  Items: '4dB/step' '2dB/step' '1dB/step' '0.5dB/step'
  Item0: '1dB/step'
Simple mixer control 'Volume Ramp Up Rate',0
  Capabilities: enum
  Items: '1 sample/update' '2 samples/update' '4 samples/update' 'Immediate'
  Item0: '1 sample/update'
Simple mixer control 'Volume Ramp Up Step',0
  Capabilities: enum
  Items: '4dB/step' '2dB/step' '1dB/step' '0.5dB/step'
  Item0: '1dB/step'

**Comment by markblinkhorn** [comment]
`rAudio1:~  #  uname -rm && amixer
6.6.16-1-rpi aarch64
Simple mixer control 'Speaker',0
  Capabilities: pvolume pswitch pswitch-joined
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 45
  Mono:
  Front Left: Playback 35 [78%] [-10.00dB] [on]
  Front Right: Playback 35 [78%] [-10.00dB] [on]
Simple mixer control 'Mic',0
  Capabilities: pvolume cvolume pswitch pswitch-joined cswitch cswitch-joined
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: Playback 0 - 37 Capture 0 - 30
  Front Left: Playback 15 [41%] [0.00dB] [off] Capture 20 [67%] [20.00dB] [on]
  Front Right: Playback 15 [41%] [0.00dB] [off] Capture 20 [67%] [20.00dB] [on]
`

`rAudio2:~  #  uname -rm && amixer
6.1.34-1-rpi-legacy-ARCH armv6l
Simple mixer control 'Bass',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 48
  Mono: 24 [50%]
Simple mixer control 'Bass Boost',0
  Capabilities: pswitch pswitch-joined
  Playback channels: Mono
  Mono: Playback [off]
Simple mixer control 'Treble',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 48
  Mono: 24 [50%]
Simple mixer control 'PCM',0
  Capabilities: pvolume pswitch pswitch-joined
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 44
  Mono:
  Front Left: Playback 21 [48%] [-20.00dB] [on]
  Front Right: Playback 21 [48%] [-20.00dB] [on]
Simple mixer control 'Auto Gain Control',0
  Capabilities: pswitch pswitch-joined
  Playback channels: Mono
  Mono: Playback [off]
`

With DAC Powered on:
`rAudio3:~  #  uname -rm && amixer
6.6.16-1-rpi aarch64
Simple mixer control 'MLC3700 AUDIO2.0OUTPUT ',0
  Capabilities: pvolume pswitch
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 64
  Mono:
  Front Left: Playback 64 [100%] [0.00dB] [on]
  Front Right: Playback 64 [100%] [0.00dB] [on]
Simple mixer control 'MLC3700 AUDIO2.0OUTPUT ',1
  Capabilities: pvolume pvolume-joined pswitch pswitch-joined
  Playback channels: Mono
  Limits: Playback 0 - 64
  Mono: Playback 64 [100%] [0.00dB] [on]
`

With DAC powered off:
`rAudio3:~  #  uname -rm && amixer
6.6.16-1-rpi aarch64
Simple mixer control 'PCM',0
  Capabilities: pvolume pvolume-joined pswitch pswitch-joined
  Playback channels: Mono
  Limits: Playback -10239 - 400
  Mono: Playback -1065 [86%] [-10.65dB] [on]
`

**Comment by markblinkhorn** [comment]
I have started a new discussion at TEAC AI-101 Audio DAC Volume always at 100% #1596 to separate this out into a separate discussion to avoid confusion here.
